### PR TITLE
Minor spelling corrections / clarifications + resolve issue #19778

### PR DESCRIPTION
**Pull request by Huizerd**
I want to clarify something about using global step with tf.contrib.model_pruning, which I think is not clearly explained (and lead to me not seeing my mistake for 3 weeks). Also, address the illegal summary names from issue #19778.

### CODE REVIEW

File: tensorflow/contrib/model_pruning/python/pruning.py [hunk 1]
```diff
@@ -520,7 +520,7 @@ def add_pruning_summaries(self):
       thresholds = get_thresholds()
       for index, mask in enumerate(masks):
```
Comment (MarkDaoust) [inline tensorflow/contrib/model_pruning/python/pruning.py]: Shouldn't this be `for mask, threshold in zip(masks,thresholds):`

(optional)

File: tensorflow/contrib/model_pruning/python/pruning.py [hunk 2]
```diff
@@ -520,7 +520,7 @@ def add_pruning_summaries(self):
...
         if not self._exists_in_do_not_prune_list(mask.name):
-          summary.scalar(mask.name + '/sparsity', nn_impl.zero_fraction(mask))
+          summary.scalar(mask.op.name + '/sparsity', nn_impl.zero_fraction(mask))
```
Comment (MarkDaoust) [inline tensorflow/contrib/model_pruning/python/pruning.py]: This _looks_ right. 

This wouldn't be broken if there was a test for it, any chance you could add one with this change so that this stays fixed?

**Review by MarkDaoust — CHANGES_REQUESTED** [review]
Docs look good.

But I don't know this code. I'd be more comfortable if we added a test at the same time.

Thanks.

File: tensorflow/contrib/model_pruning/python/pruning.py (hunk 1 above)
Comment (Huizerd) [reply tensorflow/contrib/model_pruning/python/pruning.py]: This would indeed be better, since `masks` and `thresholds` are always the same length.

File: tensorflow/contrib/model_pruning/python/pruning.py (hunk 2 above)
Comment (Huizerd) [reply tensorflow/contrib/model_pruning/python/pruning.py]: I'm pretty novice to designing tests; how would you go about doing this? I looked at the tests in [pruning_test.py](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/model_pruning/python/pruning_test.py), but didn't get very far.

**Comment by Huizerd** [comment]
The following code will give you #19778:

```python
import tensorflow as tf
import tensorflow.contrib.model_pruning as pruning

tf.reset_default_graph()

# Global step.
global_step = tf.train.get_or_create_global_step()

# Input data.
data = tf.get_variable('data', shape=[1, 10, 10, 1], initializer=tf.random_normal_initializer())

# Single conv layer with pruning mask.
outputs = 1
kernel = [2, 2]
conv = pruning.masked_conv2d(data, outputs, kernel)

with tf.Session() as sess:
    
    # Pruning hyperparameters.
    pruning_hparams = pruning.get_pruning_hparams().parse("")

    # Pruning object.
    pruning_obj = pruning.Pruning(pruning_hparams, global_step=global_step)
    
    # Pruning summaries.
    pruning_obj.add_pruning_summaries()
    
    # Summary.
    summary_writer = tf.summary.FileWriter('.', graph=sess.graph)
```

**Review by MarkDaoust — DISMISSED** [review]
I've confirmed that this update fixes the problem.

I tried turning the reproduction code into a test, but that got messy. Think we're good to go like this.

**Review by MarkDaoust — APPROVED** [review]
